tests/setup_test.go

```diff
@@ -5653,3 +5655,207 @@ func TestRunEventsNotification(t *testing.T) {
+			defer sc.stop()
+
+			giteaToken, tokenUser01 := createLinkedAccount(ctx, t, sc.gitea, sc.config)
+
+			giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+			giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			gwClient := gwclient.NewClient(sc.config.Gateway.APIExposedURL, tokenUser01)
+
+			giteaRepo, project := createProject(ctx, t, giteaClient, gwClient)
+
+			push(t, tt.config, giteaRepo.CloneURL, giteaToken, "commit", false)
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+				if err != nil {
+					return false, nil
+				}
+
+				if len(runs) == 0 {
+					return false, nil
+				}
+				if runs[0].Phase != tt.expectedRunPhase {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			runs, _, err := gwClient.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			if len(runs) != 1 {
+				t.Fatalf("expected %d run got: %d", 1, len(runs))
+			}
+
+			if runs[0].Phase != tt.expectedRunPhase {
+				t.Fatalf("expected run phase %q, got %q", tt.expectedRunPhase, runs[0].Phase)
+			}
+			if runs[0].Result != tt.expectedRunResult {
+				t.Fatalf("expected run result %q, got %q", tt.expectedRunResult, runs[0].Result)
+			}
+
+			_ = testutil.Wait(30*time.Second, func() (bool, error) {
+				combinedStatus, _, err := giteaClient.GetCombinedStatus(agolaUser01, giteaRepo.Name, "master")
+				if err != nil {
+					return false, nil
+				}
+
+				if combinedStatus.State != tt.expectedGiteaStatusState {
+					return false, nil
+				}
+
+				return true, nil
+			})
+
+			targetURL, err := webRunURL(sc.config.Notification.WebExposedURL, project.ID, runs[0].Number)
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+
+			combinedStatus, _, err := giteaClient.GetCombinedStatus(agolaUser01, giteaRepo.Name, "master")
+			if err != nil {
+				t.Fatalf("unexpected err: %v", err)
+			}
+			if combinedStatus.State != tt.expectedGiteaStatusState {
+				t.Fatalf("expected commit state %q, got %q", tt.expectedGiteaStatusState, combinedStatus.State)
+			}
+			if combinedStatus.Statuses[0].Description != tt.expectedGiteaDescription {
+				t.Fatalf("expected commit state description %q, got %q", tt.expectedGiteaDescription, combinedStatus.Statuses[0].Description)
+			}
+			if combinedStatus.Statuses[0].Context != tt.expectedGiteaContext {
+				t.Fatalf("expected commit state context %q, got %q", tt.expectedGiteaContext, combinedStatus.Statuses[0].Context)
+			}
+			if combinedStatus.Statuses[0].TargetURL != targetURL {
+				t.Fatalf("expected commit state targetURL %q, got %q", targetURL, combinedStatus.Statuses[0].TargetURL)
+			}
+		})
+	}
+}
+
+func webRunURL(webExposedURL, projectID string, runNumber uint64) (string, error) {
```

Comment:
This function has been copied from the notification service. Define it inline inside the test instead of globally and add a comment that it has been copied from the notification service.